src/components/App.js

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import Search from './Search';
+import Results from './Results';
+import Detail from './Detail';
+import Favourites from './Favourites';
+import cx from 'classnames';
+
+class App extends React.Component {
+  
+  constructor() {
+    super();
+    this.state = { submittedQuery: '',
+                   previewQuery: '',
+                   results: [],
+                   detail: '',
+                   preview: [],
+                   favourites: [],
+                   page: 1,
+                   newSearchInProgress: false,
+                   nextPageFetchInProgress: false,
+                   resultsDisplayed: false,
+                   detailDisplayed: false,
+                   favsDisplayed: false, 
+                   movieNotFound: false };
+
+    this.receiveSearch = this.receiveSearch.bind(this);
+    this.getDetail = this.getDetail.bind(this);
+    this.closeDetail = this.closeDetail.bind(this);
+    this.addFav = this.addFav.bind(this);
+    this.removeFav = this.removeFav.bind(this);
+    this.handleFavMenu = this.handleFavMenu.bind(this);
+    this.handleLogout = this.handleLogout.bind(this);
+    this.receiveMovie = this.receiveMovie.bind(this);
+    this.moveFavUp = this.moveFavUp.bind(this);
+    this.moveFavDown = this.moveFavDown.bind(this);
+  }
+
+  componentDidMount() {
+    this.setState({favourites: JSON.parse(window.localStorage.favourites)});
+  }
+
+  receiveSearch(query, type) {
+
+    let apiQuery;
+
+    if (type === 'results') {
+      this.setState({ submittedQuery: query,
+                      page: 1,
+                      newSearchInProgress: true });
+      apiQuery = `http://www.omdbapi.com/?apikey=eee5954b&s=${query}&type=movie`;
+      this.fetchData(apiQuery, type);
+    }
+
+    else if (type === 'resultsInfinite') {
+      const {nextPageFetchInProgress} = this.state;
+      if (!nextPageFetchInProgress) {
+        const page = this.state.page + 1;
+        this.setState({ page,
+                        nextPageFetchInProgress: true });
+        apiQuery = `http://www.omdbapi.com/?apikey=eee5954b&s=${query}&page=${page}&type=movie`;
+        this.fetchData(apiQuery, type);
+      }
+    }
+
+    else if (type === 'preview') {
+      this.setState({previewQuery: query}); 
+      apiQuery = `http://www.omdbapi.com/?apikey=eee5954b&s=${query}&type=movie`;
+      this.fetchData(apiQuery, type);
+    }
+  }
+
+  receiveMovie(movie, source) {
+    const apiQuery = `http://www.omdbapi.com/?apikey=eee5954b&i=${movie.imdbID}`;
+    this.fetchData(apiQuery, 'detail');
+    if (source === 'fromFavs') {
+      setTimeout(() => this.setState({ favsDisplayed: false }), 500);
```

Comment:
setTimeout feels awkward here. there is probably a better way of doing this. give me a shout to go over this if you want

Reply:
probably css transitions. was too lazy to deal with css on the weekend :)